product_abc_classification_sale_stock/tests/test_abc_classification_profile.py

```diff
@@ -275,9 +275,15 @@ def test_02(self):
         self.assertFalse(levels.sale_stock_level_history_ids)
         self.stock_profile._compute_abc_classification()
         levels = self.product1.abc_classification_product_level_ids
-        self.assertEqual(len(levels.sale_stock_level_history_ids), 1)
+        logs = self.env["abc.sale_stock.level.history"].search(
+            [("product_level_id", "in", levels.ids)]
+        )
+        self.assertEqual(len(logs), 1)
```

Comment:
This should not be changed. The `sale_stock_level_history_ids` should have been invalidated by the call to `_compute_abc_classification` This is the purpose of the initial assert. That means that the invalidation done into the `_log_history` method is not complete and we should add

```
self.env["product.template"].invalidate_cache(
            ["abc_classification_product_level_ids"]
        )
self.env["product.product"].invalidate_cache(
            ["abc_classification_product_level_ids"]
        )
```

Reply:
I tried changes, because module testing failed in the last test, the reason why I tested version 16.0 is that refrsh() method is deprecated in version 16.0 but I didn't see the warning in the log for version 17.0.

I preferred to use one of the recommended three methods that I saw in the log of version 16.0.

Even this method:
self.env["product.template"].invalidate_cache(
            ["abc_classification_product_level_ids"]
        )
 is deprecated in version 16.0 and raising errors in version 17.0 without giving warnings in the log, so shall we override deprecated methods and see how can it work with the new not deprecated ones?
 
 Also the database column is changed by the pre-commit command, this one if I return to the original name can the levels be computed automatically?
 
 For now, let me remove my name from maintainers and return the old column name in the DB, but will need your help for the remaining ones as I inquired above.

Reply:
To be clear the history records were always equal to 0, and assertions should be 1==1, and 2==2.

The original 16.0 version had two issues:
1 deprecated methods for migration to 17.0
2 the last test for logging history

Reply:
![Screenshot from 2024-12-10 12-46-01](https://github.com/user-attachments/assets/a36b542f-760f-4f02-b105-878ac39b94b2)